### PR TITLE
#160710724  Enable Travel Administrator to View Further Guest House Details 

### PR DESCRIPTION
**Pull request by johnmutuma5**
#### What does this PR do?
- Enable admin to view further details and a timeline for a room's bed bookings

#### Description of Task to be completed?
- Fetch further guest house details: be able to get the number of rooms, vacant spaces
- Display the information on a timeline
  - Enable the user to navigate time on the timeline
  - Enable the user to view the timeline in `Weeks`, `Months` or `Years`
  - Enable the user to click on the trips in the timeline to view further trip information
 
#### How should this be manually tested?
- Clone the repository and enter the project's folder
- Checkout to this branch and pull the branch
- Assign yourself the travel admin role by changing your roleId to 29187 in the users table
- Start the application using 'make start'

##### Adding data
- Open postman and provide your token in the authorization header. 
- Create a guesthouse with some rooms and beds via` POST to 127.0.0.1:5000/api/v1/guesthouses/`
   - You can use this [sample data](https://gist.github.com/johnmutuma5/ef11b9b3455ffc7be83111dcc2f41ab1) to create a guest house
- Get the `bedIds` in the response and use them to create some test requests in the database. You may also find this [sample dataa](https://gist.github.com/johnmutuma5/696ecf6ad7bc65f6ea7aa0d3792055d3) for a request useful. Ensure the `bedId`s are existent in the Beds database table
- Set up the backend for responses and view the request created on the UI, feel free to add as many trips as you'd like making sure that the dates don't collide for bed's. See below on how to set up the backend;

###### Setting up the backed
- Follow the description on [this](https://github.com/andela/travel_tool_back/pull/78) pull request to set up the backed
 
#### What are the relevant pivotal tracker stories?
#160710724
#### Screenshots (if appropriate)
screen shot 2018-10-04 at 05 36 19

screen shot 2018-10-03 at 23 11 25

### CODE REVIEW

**Comment by samuelkubai** [comment]
@johnmutuma5 please fix this title.

**Comment by johnmutuma5** [comment]
@samuelkubai, working on it and the description too.

File: src/components/CentreGrid/CentreCard.jsx [hunk 1]
```diff
@@ -88,7 +92,8 @@ CentreCard.propTypes = {
   guestHouseName: PropTypes.string.isRequired,
   guestHouseLocation: PropTypes.string.isRequired,
   beds: PropTypes.number.isRequired,
-  bathrooms: PropTypes.number.isRequired
+  bathrooms: PropTypes.number.isRequired,
+  guesthouseId: PropTypes.number.isRequired
```
[the remark this answers is not in the document]
Comment (johnmutuma5) [reply src/components/CentreGrid/CentreCard.jsx]: @d3mola, it's a string. Changing that.

**Comment by d3mola** [comment]
screen shot 2018-10-04 at 2 17 42 pm

That icon looks weird on my end, am I missing something?

File: src/components/GuestHouseDetailCard/GuestHouseDetailCard.scss [hunk 2]
```diff
@@ -0,0 +1,33 @@
+.guesthouse-detail-card {
+  display: flex;
+  flex-flow: column;
+  justify-content: center;
+  align-items: center;
+  min-width: 18%;
+  height: 120px;
+  padding: 10px 20px;
+  border: 1px solid #E4E4E4;
+	border-radius: 2px;
```
[the remark this answers is not in the document]
Comment (johnmutuma5) [reply src/components/GuestHouseDetailCard/GuestHouseDetailCard.scss]: Thanks @d3mola, checking it out.

[the remark this answers is not in the document]
Comment (johnmutuma5) [reply src/components/GuestHouseDetailCard/GuestHouseDetailCard.scss]: @d3mola, the indentation looks okay on the IDE, could Github be pushing it to the left?

**Comment by johnmutuma5** [comment]
> screen shot 2018-10-04 at 2 17 42 pm
> 
> That icon looks weird on my end, am I missing something?

@d3mola, fixing that too.

File: src/components/GuestHouseDetailCard/GuestHouseDetailCard.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (d3mola) [reply src/components/GuestHouseDetailCard/GuestHouseDetailCard.scss]: You're right, it looks alright in my text editor too

**Comment by d3mola** [comment]
screen shot 2018-10-04 at 3 07 32 pm

@johnmutuma5 there should be some padding above `SMALL CUTTER`

**Comment by samuelkubai** [comment]
@johnmutuma5 please resolve conflicts.

**Comment by samuelkubai** [comment]
@johnmutuma5 please resolve conflicts.

**Comment by johnmutuma5** [comment]
Great @daddychukz. Thanks a bunch for the reviews.

**Pull request merged**